### PR TITLE
Adaptivity fixes

### PR DESCRIPTION
**Pull request by gpolitis**

### CODE REVIEW

File: src/main/java/org/jitsi/videobridge/cc/BitrateController.java [hunk 1]
```diff
@@ -704,14 +723,16 @@ private EndpointBitrateAllocation(
...
         void allocate(long maxBps, int maxQuality)
         {
-            if (!forwarded || rates.length == 0)
+            if (rates.length == 0)
             {
                 return;
             }
 
-            maxQuality = selected ? Math.min(maxQuality, rates.length - 1) : 0;
+            maxQuality = Math.min(maxQuality, optimalIdx);
```
Comment (bbaldino) [inline src/main/java/org/jitsi/videobridge/cc/BitrateController.java]: does `maxQuality` refer to an index?  i think it'd be good to rename it to `maxQualityIdx`

File: src/main/java/org/jitsi/videobridge/xmpp/MediaStreamTrackFactory.java [hunk 2]
```diff
@@ -113,7 +113,9 @@
         // scalability (VP9). Exotic cases might do simulcast + spatial
         // scalability.
 
+        int resolution = 180;
```
Comment (bbaldino) [inline src/main/java/org/jitsi/videobridge/xmpp/MediaStreamTrackFactory.java]: use the thumbnail resolution constant here?

Comment (gpolitis) [inline src/main/java/org/jitsi/videobridge/xmpp/MediaStreamTrackFactory.java]: This `resolution` should, in theory, be included in the SDP answer of a sender, while the thumbnail resolution constraint in the bitrate controller should come from the receiver as a data channel message (as it is requested here https://github.com/jitsi/jitsi-videobridge/issues/147) or even in the answer of a receiver.

So these two refer to different things and I feel uncomfortable using the same constant. How about I use another constant here (and I also include the above explanation)?

**Comment by gpolitis** [comment]
@bbaldino thanks! Let's wait for a new LJ build and update the pom.xml file (otherwise the project won't build).

File: src/main/java/org/jitsi/videobridge/cc/BitrateController.java [hunk 3]
```diff
@@ -685,12 +692,24 @@ private EndpointBitrateAllocation(
...
             {
                 rates[i] = encodings[i].getLastStableBitrateBps();
+                if (encodings[i].getHeight() <= THUMBNAIL_MAX_HEIGHT)
+                {
+                    optimalThumbnailIndex = i;
+                }
             }
 
-            optimalIdx = selected ? encodings.length - 1 : (forwarded ? 0 : -1);
+            // TODO Determining the optimal index needs some work. The optimal
```
Comment (saghul) [inline src/main/java/org/jitsi/videobridge/cc/BitrateController.java]: I'm not sure we'd ever want to do that. Mobile devices these days have so many pixels that a 360p video would look terrible.

**Pull request merged**